addons/dataverse/models.py

```diff
@@ -33,25 +31,11 @@ def _hashes(self):
             return None
 
     def update(self, revision, data, save=True, user=None):
-        """Note: Dataverse only has psuedo versions, pass None to not save them
+        """Note: Dataverse only has psuedo versions (_history), pass None to not save them
         Call super to update _history and last_touched anyway.
-        Dataverse requires a user for the weird check below
         """
-        version = super(DataverseFile, self).update(None, data, user=user, save=save)
+        version = super().update(None, data, user=user, save=save)
         version.identifier = revision
-
-        user = user or _get_current_user()
-        if not user or not self.target.has_permission(user, WRITE):
-            try:
-                # Users without edit permission can only see published files
-                if not data['extra']['hasPublishedVersion']:
-                    # Blank out name and path for the render
-                    # Dont save because there's no reason to persist the change
-                    self.name = ''
-                    self.materialized_path = ''
```

Comment:
I assume this was getting saved as blank and that was leading to the 500s

Reply:
The 500s were shown on Sentry to be not Keyerror related, here's one: https://staging-sentry.cos.io/cos/osf-back-end-77/issues/13311/ I'm guessing that this happens if multiple filenodes get saved with materialized_path == ''. @felliott was looking into this yesterday.